shared/wallets/send-form/asset-input/asset-input-advanced.tsx

```diff
@@ -0,0 +1,245 @@
+      direction="vertical"
+      fullWidth={true}
+      style={Styles.collapseStyles([sharedStyles.container, styles.container])}
+      gap="xtiny"
+    >
+      <Kb.Box2 direction="horizontal" fullWidth={true} style={styles.topLabel}>
+        {buildingAdvanced.recipientType === 'keybaseUser' ? (
+          <>
+            <Kb.Avatar username={buildingAdvanced.recipient} size={16} style={styles.avatar} />
+            <Kb.ConnectedUsernames
+              usernames={[buildingAdvanced.recipient]}
+              type="BodyTinySemibold"
+              colorBroken={true}
+              colorFollowing={true}
+              underline={false}
+            />
+          </>
+        ) : (
+          <Kb.Text type="BodyTinySemibold" lineClamp={1} ellipsizeMode="middle">
+            {buildingAdvanced.recipient}
+          </Kb.Text>
+        )}
+        <Kb.Text type="BodyTinySemibold" style={styles.noShrink}>
+          will receive:
+        </Kb.Text>
+      </Kb.Box2>
+      <Kb.Box2 direction="horizontal" fullWidth={true}>
+        <AmountInput
+          numDecimalsAllowed={7}
+          onChangeAmount={onChangeAmount}
+          value={buildingAdvanced.recipientAmount}
+        />
+        <PickAssetButton isSender={false} />
+      </Kb.Box2>
+    </Kb.Box2>
+  )
+}
+
+const LeftBlock = (props: EmptyProps) => {
+  const buildingAdvanced = Container.useSelector(state => state.wallets.buildingAdvanced)
+  const builtPaymentAdvanced = Container.useSelector(state => state.wallets.builtPaymentAdvanced)
+  return builtPaymentAdvanced.sourceDisplay ? (
+    <Kb.Box2 direction="vertical" alignItems="flex-start">
+      <Kb.Text type="HeaderBigExtrabold" style={!!builtPaymentAdvanced.amountError && styles.error}>
+        ~{builtPaymentAdvanced.sourceDisplay}
+      </Kb.Text>
+      <Kb.Text type="BodyTiny">At most {builtPaymentAdvanced.sourceMaxDisplay}</Kb.Text>
+      {!!buildingAdvanced.recipientAsset && (
+        <Kb.Text type="BodyTiny">{builtPaymentAdvanced.exchangeRate}</Kb.Text>
+      )}
+      {!!builtPaymentAdvanced.amountError && (
+        <Kb.Text type="BodySmall" style={styles.error}>
+          {builtPaymentAdvanced.amountError}
+        </Kb.Text>
+      )}
+    </Kb.Box2>
+  ) : (
+    <CalculateAdvancedButton isIcon={true} />
+  )
+}
+
+export const AssetInputSenderAdvanced = (props: EmptyProps) => (
+  <Kb.Box2
+    direction="vertical"
+    fullWidth={true}
+    style={Styles.collapseStyles([sharedStyles.container, styles.container])}
+  >
+    <Kb.Text type="BodyTinySemibold" style={styles.topLabel}>
+      You will send approximately:
+    </Kb.Text>
+    <Kb.Box2 direction="horizontal" fullWidth={true}>
+      <LeftBlock />
+      <Kb.Box style={Styles.globalStyles.flexGrow} />
+      <PickAssetButton isSender={true} />
+    </Kb.Box2>
+    <Available />
+  </Kb.Box2>
+)
+
+export const AssetPathIntermediate = () => {
+  const path = Container.useSelector(state => state.wallets.builtPaymentAdvanced.fullPath.path)
```

Comment:
seeing deep chaining without guards is a little worrying.. any chance that this part of the store disappears from under the component, e.g. by being cleared out by a new payment being prepared in another nav tab, or the info being cleared out immediately after sending?

Reply:
Hmm I don't think so. All the links in this path are populated. They are just empty records or strings when not initialized or cleared.